clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -323,11 +323,17 @@ private synchronized void disableHeartbeatThread() {
     }
 
     private void closeHeartbeatThread() {
```

Comment:
It's called through `AbstractCoordinator#close`.  I didn't track down all the callers of that function.  In any case, just fixing it seems easier than trying to document the lack of thread-safety, and this is not performance-critical.